homeassistant/components/tplink/const.py

```diff
@@ -22,3 +23,40 @@
 CONF_DEVICE_CONFIG: Final = "device_config"
 
 PLATFORMS: Final = [Platform.LIGHT, Platform.SENSOR, Platform.SWITCH]
+
+
+class EntityExtras(TypedDict):
+    """Class to define additional properties to be set on feature based entities."""
+
+    key: str | None
+    device_class: str | None
+    state_class: str | None
+
+
+ENTITY_EXTRAS: Final = {
```

Comment:
Wrt the comment from Nick, perhaps we should move extras to be kept in their respective platform files?

As we probably want to generalize accesses for this, perhaps a class variable inside the platform class should alias the extras? So something like:
```
SENSOR_ENTITY_EXTRAS: Final = {

}

class Sensor(...):
    _entity_extras = SENSOR_ENTITY_EXTRAS
```
Just a thought.

Reply:
Now done in https://github.com/home-assistant/core/pull/120025